src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Agent/AgentResponseParserHelpers.cs

```diff
@@ -0,0 +1,146 @@
+using System.Text.Json;
+using Azure.Tools.GeneratorAgent.Models;
+using Azure.Tools.ErrorAnalyzers;
+
+namespace Azure.Tools.GeneratorAgent.Agent
+{
+    /// <summary>
+    /// Static helper methods for parsing AI agent responses in plain text JSON format
+    /// </summary>
+    internal static class AgentResponseParserHelpers
+    {
+        /// <summary>
+        /// Parses the agent's response and extracts the updated client.tsp content from JSON
+        /// </summary>
+        public static AgentTypeSpecResponse ParseResponse(string rawResponse)
+        {
+            if (string.IsNullOrWhiteSpace(rawResponse))
+            {
+                throw new ArgumentException("Agent response is empty or null", nameof(rawResponse));
+            }
+
+            try
+            {
+                // Extract JSON from markdown code blocks if present
+                var jsonContent = ExtractJsonFromResponse(rawResponse);
+
+                // Parse as JSON
+                var response = JsonSerializer.Deserialize<AgentTypeSpecResponse>(jsonContent, new JsonSerializerOptions
```

Comment:
Should we create a static instance of these options, since they're immutable and we're not changing the values per call?